kqueen/auth/common.py

```diff
@@ -9,21 +9,66 @@
 import importlib
 import logging
 
+config = current_config()
 logger = logging.getLogger('kqueen_api')
 
+"""
+Authentication Modules
+
+To define new module, need to specify it as dictionary, where:
+
+"auth_option_lower_case": {"engine": "EqualsToAuthClassName",
+                           "param": {
+                               "key": "value"
+                           }
+                           }
+"""
+
+AUTH_MODULES = {
+    "ldap": {
+        "engine": "LDAPAuth",
+        "param": {
+            "uri": config.get('LDAP_URI')
+        }
+    },
+    "local": {
+        "engine": "LocalAuth",
+        "param": {
+
+        }
+    }
+}
+
+
+def generate_auth_options(auth_list):
+    auth_options = {}
+
+    methods = auth_list.strip().split(',')
+    for m in methods:
+        if AUTH_MODULES[m]:
+            auth_options[m] = AUTH_MODULES[m]
+
+    if not auth_options:
+        auth_options['local'] = {'engine': 'LocalAuth', 'param': {}}
+
+    logger.debug('Auth config generated {}'.format(auth_options))
+    return auth_options
+
 
 def get_auth_instance(name):
     # Default type is local auth
 
     config = current_config()
-    auth_config = config.get("AUTH", {}).get(name, {})
+    auth_config = generate_auth_options(config.get("AUTH_MODULES")).get(name, {})
```

Comment:
Maybe split this into two lines just to improve the readability a bit:
```
    auth_options = generate_auth_options(config.get('AUTH_MODULES'))
    auth_config = auth_options.get(name, {})
```

Reply:
okko